apps/OboeTester/app/src/main/cpp/NativeAudioContext.cpp

```diff
@@ -358,6 +360,33 @@ double ActivityContext::getTimestampLatency(int32_t streamIndex) {
     return -1.0;
 }
 
+void ActivityContext::stopBlockingIOThread() {
+    if (dataThread != nullptr) {
+        // stop a thread that runs in place of the callback
+        threadEnabled.store(false); // ask thread to exit its loop
+        std::shared_ptr<oboe::AudioStream> oboeStream = getOutputStream();
+        if (oboeStream != nullptr &&
+            oboeStream->getPerformanceMode() == PerformanceMode::POWER_SAVING_OFFLOADED) {
+            std::lock_guard _l(threadLock);
```

Comment:
Will we get into trouble with both threadEnabled and threadLock? Is there a better way solve this problem with less variables?